django_elasticsearch_dsl/documents.py

```diff
@@ -202,13 +202,26 @@ def _prepare_action(self, object_instance, action):
         }
 
     def _get_actions(self, object_list, action):
-        if self._doc_type.queryset_pagination is not None:
-            paginator = Paginator(
-                object_list, self._doc_type.queryset_pagination
-            )
-            for page in paginator.page_range:
-                for object_instance in paginator.page(page).object_list:
-                    yield self._prepare_action(object_instance, action)
+        qs_pagination = self._doc_type.queryset_pagination
+        if qs_pagination is not None:
+            if isinstance(object_list, models.QuerySet):
+                # Django paginator can lead to performance issue with huge
+                # queryset, make custom one based on pks presents in db right
+                # now
+                pks = list(object_list.values_list('pk', flat=True))
```

Comment:
I can't really test it. Do you have tested to retrieve millions of pks in this list ?
And don't you think that this `object_list.values_list` queryset should be ordered by pk ?

Reply:
Of course ! improve this code with the new commit (--amend)